### PR TITLE
sitewise sdk

### PR DESCRIPTION
**Pull request by tracfren**
This change is the initial introduction of the AWS SDK SiteWise client.

I also ran prettier on the whole code base, so you'll see a lot of extra changes.

### CODE REVIEW

File: src/application/index.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { IoTSiteWiseClient } from "@aws-sdk/client-iotsitewise";
```
Comment (NorbertNader) [inline src/application/index.ts:1]: why not just call this directory client?

File: src/entities/SiteWise/Asset.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import type { DescribeAssetCommandOutput } from "@aws-sdk/client-iotsitewise";
```
Comment (NorbertNader) [inline src/entities/SiteWise/Asset.ts:1]: Why is this directory called entitites?

File: src/data/SiteWise/IoTSiteWiseGateway.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { IoTSiteWiseClient } from "@aws-sdk/client-iotsitewise";
+
+import DescribeAsset from "./DescribeAsset";
+import ListAssets from "./ListAssets";
+import ListAssociatedAssets from "./ListAssociatedAssets";
+
+class IoTSiteWiseGateway {
+  private client: IoTSiteWiseClient;
+  private _describeAsset: DescribeAsset;
```
Comment (NorbertNader) [inline src/data/SiteWise/IoTSiteWiseGateway.ts:9]: why the underscores?

File: src/application/index.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import IoTSiteWiseGateway from "../data/SiteWise";
+
+const client = new IoTSiteWiseClient({
+  region: "us-west-2",
+  credentials: {
+    secretAccessKey: "",
+    accessKeyId: "",
```
Comment (NorbertNader) [inline src/application/index.ts:8]: don't we also need a sessionToken?

File: src/data/SiteWise/ListAssociatedAssets.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    this.client = client;
+  }
+
+  public async invoke(
+    assetId: string,
+    hierarchyId: string,
+  ): Promise<AssetSummary[]> {
+    try {
+      const assetSummaries: AssetSummary[] = [];
+      for await (const page of paginateListAssociatedAssets(
```
Comment (square-li) [inline src/data/SiteWise/ListAssociatedAssets.ts:22]: Would `Promise.all()` works better here? `Promise.all()` invokes all promises but for await only invokes one at a time. Does order matters here?

**Comment by NorbertNader** [comment]
At a higher level, why do we need sdk in application? Won't iot-app-kit handle any calls on the client?

**Comment by tracfren** [comment]
I'm closing this review as part of a refactor I am doing. 

@NorbertNader you have a great point. I have a feeling we will need it, but you're right, there isn't a reason for it at this time. let's delay adding it until we need it.

**Pull request closed without merging**